deepchem/models/torch_models/modular.py

```diff
@@ -342,3 +271,127 @@ def fit_generator(self,
         time2 = time.time()
         logger.info("TIMING: model fitting took %0.3f s" % (time2 - time1))
         return last_avg_loss
+
+    def load_from_pretrained(  # type: ignore
+            self,
+            source_model: Optional["ModularTorchModel"] = None,
```

Comment:
Are all three required - source_model, checkpoint, model_dir? I think only model_dir will be sufficient. Given a model_dir, the method loads the state_dict and if any of the current models layer or component matches the keys in state_dict, the method can update those components weights.

Reply:
I think it's relatively harmless to support multiple loading options here. Gives maybe a bit more flexibility to users